### PR TITLE
Swift 4

### PR DESCRIPTION
**Pull request by klefevre**
@jeremiegirault could you review if I missed something on this one please ?

### CODE REVIEW

**Comment by jeremiegirault** [comment]
Sure I'm going to look at it

File: Postal/Postal.swift [hunk 1]
```diff
@@ -268,7 +268,7 @@ public extension Postal {
 
 private extension Postal {
     
-    func doAsync<T, E: Error>(_ f: @escaping () throws -> T, completion: @escaping (Result<T, E>) -> Void) {
+    func doAsync<T, E>(_ f: @escaping () throws -> T, completion: @escaping (Result<T, E>) -> Void) {
```
Comment (jeremiegirault) [inline Postal/Postal.swift]: was it intended ?

**Comment by jeremiegirault** [comment]
BTW I guess you can update the README badge for swift version :)

File: Postal/Postal.swift (hunk 1 above)
Comment (klefevre) [reply Postal/Postal.swift]: It's to fix a warning `Redundant conformance constraint 'E': 'Error'`.

Comment (jeremiegirault) [reply Postal/Postal.swift]: Because result already constraints `E: Error` ? Interesting, did not know that feature of swift 4. Thank you !

Comment (klefevre) [reply Postal/Postal.swift]: Me neither, was surprised actually.

Comment (jeremiegirault) [reply Postal/Postal.swift]: What's surprising is that it did not trigger in some other places, but we did not use a lot of generic constraints in Postal

**Comment by klefevre** [comment]
Well I think we're good to go.

**Pull request merged**